Add files via upload

Doraz1 opened the pull request:
Just

 created the forwards propagation of a CNN. Spent most of my time reading up on them and I think I get the point. Shouldn't take too long to create the backpropagation and then move on to the evolution engine that takes a generation of nets with fitnesses and creates the next generation out of them.

Doraz1 committed:
Add files via upload

Just created the forwards propagation of a CNN. Spent most of my time reading up on them and I think I get the point. Shouldn't take too long to create the backpropagation and then move on to the evolution engine that takes a generation of nets with fitnesses and creates the next generation out of them.

gthecht commented on main.py:
```diff
@@ -0,0 +1,138 @@
+#
+#         # TODO : create a net with these new weights and return the net
+#         return new_weights
+
```
These can be deleted...

gthecht commented:
In your commit, just write what is in the commit - what you added and what you removed.
For instance:
Adds EvolutionEngine class that evolves the best networks.

gthecht commented on main.py:
```diff
@@ -0,0 +1,138 @@
+
+        while len(self.next_gen) < POP_SIZE:
+            index = random.randint(num_survivors)
+            mutated_offspring = self.mutate(survivor_list[index])
```
What is self.mutate?

Doraz1 replied:
This will take a net that passes the survival threshold, mutate its hyperparameters and add the offspring to the next generation

Doraz1 committed:
Add files via upload

Adds ConvolutionLayer class that receives an input and propagates it forwards, as well as calculates all gradients from weights and from inputs to outputs
Adds Net class that holds all ConvolutionLayers and propagates an input through them, returning the net output at the final layer.

Doraz1 committed:
Merge pull request #2 from gthecht/Doraz1-patch-2

Add files via upload
